pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -3925,6 +3925,132 @@ func TestReconcilePipeline_TaskSpecMetadata(t *testing.T) {
 	}
 }
 
+func TestReconcileWithTaskResultsInFinalTasks(t *testing.T) {
+	names.TestingSeed()
+	ps := []*v1beta1.Pipeline{tb.Pipeline("test-pipeline", tb.PipelineNamespace("foo"), tb.PipelineSpec(
+		tb.PipelineTask("dag-task-1", "dag-task"),
+		tb.PipelineTask("dag-task-2", "dag-task"),
+		tb.FinalPipelineTask("final-task-1", "final-task-1",
+			tb.PipelineTaskParam("finalParam", "$(tasks.dag-task-1.results.aResult)"),
+		),
+		tb.FinalPipelineTask("final-task-2", "final-task-2",
+			tb.PipelineTaskParam("finalParam", "$(tasks.dag-task-2.results.aResult)"),
+		),
+	))}
```

Comment:
I believe we are trying to deprecate test builders from our tests. :sweat_smile: #3178 
It's not a blocker. I would like to lgtm this. I am glad to make this change later, if you want to merge this now.

Reply:
thanks @popcor255 yes, will change that later along with the other tests.

Reply:
pinging this again

Reply:
🤦‍♀️  shying away in this PR